instaclustr/resource_firewall_rule.go

```diff
@@ -53,11 +57,23 @@ func resourceFirewallRuleCreate(d *schema.ResourceData, meta interface{}) error
 
 		rules = append(rules, RuleType{Type: aRule})
 	}
-
-	rule := FirewallRule{
-		Network: d.Get("rule_cidr").(string),
-		Rules:   rules,
+	var rule FirewallRule
```

Comment:
[Desired, simplicity] - using return early return often + merging 2 blocks, could this logic be simplified to 

```
if d.Get("rule_cidr") != "" && d.Get("rule_security_group_id") != "" {
		return fmt.Errorf("[Error] Error creating firewall rule: Only one of Security Group of Rule Cidr can be provided per rule")
	}

	if d.Get("rule_cidr") == "" && d.Get("rule_security_group_id") == "" {
		return fmt.Errorf("[Error] Error creating firewall rule: either one of Security Group of Rule Cidr is required")
	}

	var rule = FirewallRule{
		Network:         d.Get("rule_cidr").(string),
		SecurityGroupId: d.Get("rule_security_group_id").(string),
		Rules:           rules,
	}
````

Reply:
I will check if that works - if it does then i will do this :)